testsuite/utils.py

```diff
@@ -170,17 +170,24 @@ def _asdict_recurse(obj):
     return result
 
 
+def check_condition(condition, condition_type, status, reason, message):
+    """Checks if condition matches expectation"""
```

Comment:
maybe document that when message or reason are None they are ignored during resolution